matplotcheck/autograde.py

```diff
@@ -0,0 +1,89 @@
+# Wrapper functions for matplotcheck functions that run
+# and track the passing status of test functions and
+# provide formatted results of tests
+
+
+def run_test(
+    func,
+    points,
+    *args,
+    correct_message="default correct",
+    error_message="default error",
+    **kwargs
+):
+    """Runs a pre-defined test function and creates a dictionary
+    containing the results of the test
+
+    Parameters
+    ---------
+    func: name of pre-defined test function to run
+    points: number of points assigned for passing test
+    args: arguments provided to test function
+    correct_message: string of custom message returned with passing test
+    error_message: string of custom message returned with failing test
+    kwargs: keyword arguments provided to test function
+
+    Returns
+    -------
+    results: dictionary with the following key:value pairs
+        points: number of points assigned based on test results
+        pass: boolean of passing status of test
+        description: string of test function name that was run
+        message: string of custom message returned based on passing status
+        [correct_message or error_message]
+        traceback: error message from test function (when pass is False)
+    """
+    results = {"points": 0, "pass": False}
+    score = 0
+    try:
+        fname = func.__name__
+        results["description"] = fname
+        func(*args, **kwargs)
+    except Exception as e:
+        results["message"] = error_message
+        results["traceback"] = e
+        pass
+    else:
+        results["pass"] = True
+        results["message"] = correct_message
+        results["points"] = points
+
+    return results
+
+
+def output_results(results):
+    """Prints a formatted message containing the total number of points
+    summed across a list of dictionaries with results from one or more tests
+
+    Parameters
+    ---------
```

Comment:
this line also looks short. i'm surprised this rendered as normally that triggers an error with rst

Reply:
Added the missing `-`.